### PR TITLE
feat: Implement a wildcard in http allowed_domains to allow all domains

### PR DESCRIPTION
**Pull request by Mossaka**
This PR fixes the issue #421 . 

It does 
1. add a wildcard `"*"` option to `allowed_http_hosts` in HTTP component. 
2. add an example to demonstrate how to use it to make outbound http call in rust

Note: this PR has a dependency on https://github.com/deislabs/wasi-experimental-http/pull/91. <del>Once the upstream PR is merged, this PR will go off-draft. </del>

### CODE REVIEW

File: crates/outbound-http/Cargo.toml [hunk 1]
```diff
@@ -19,5 +19,6 @@ tokio = { version = "1.4.0", features = [ "full" ] }
 tracing = { version = "0.1", features = [ "log" ] }
 tracing-futures = "0.2"
 url = "2.2.1"
-wasi-experimental-http-wasmtime = { git = "https://github.com/deislabs/wasi-experimental-http", rev = "4ed321d6943f75546e38bba80e14a59797aa29de" }
-wit-bindgen-wasmtime = { git = "https://github.com/bytecodealliance/wit-bindgen", rev = "2f46ce4cc072107153da0cefe15bdc69aa5b84d0" }
\ No newline at end of file
+wasi-experimental-http-wasmtime = { git = "https://github.com/Mossaka/wasi-experimental-http", rev = "c09372349dadf3a6d5b5fdad5ae0d89e2daad19b" }
+wit-bindgen-wasmtime = { git = "https://github.com/bytecodealliance/wit-bindgen", rev = "2f46ce4cc072107153da0cefe15bdc69aa5b84d0" }
+log = "0.4"
```
[the remark this answers is not in the document]
Comment (Mossaka) [reply crates/outbound-http/Cargo.toml:24]: thanks!

File: crates/outbound-http/src/lib.rs [hunk 2]
```diff
@@ -35,13 +36,20 @@ impl OutboundHttp {
...
         match allowed_hosts.as_deref() {
             Some(domains) => {
-                let allowed: Result<Vec<_>, _> = domains.iter().map(|d| Url::parse(d)).collect();
-                let allowed = allowed.map_err(|_| HttpError::InvalidUrl)?;
-
-                Ok(allowed
-                    .iter()
-                    .map(|u| u.host_str().unwrap())
-                    .any(|x| x == url_host.as_str()))
+                log::info!("Allowed hosts: {:?}", domains);
```
[the remark this answers is not in the document]
Comment (Mossaka) [reply crates/outbound-http/src/lib.rs:39]: changed

File: crates/outbound-http/Cargo.toml [hunk 3]
```diff
@@ -19,5 +19,5 @@ tokio = { version = "1.4.0", features = [ "full" ] }
 tracing = { version = "0.1", features = [ "log" ] }
 tracing-futures = "0.2"
 url = "2.2.1"
-wasi-experimental-http-wasmtime = { git = "https://github.com/deislabs/wasi-experimental-http", rev = "4ed321d6943f75546e38bba80e14a59797aa29de" }
-wit-bindgen-wasmtime = { git = "https://github.com/bytecodealliance/wit-bindgen", rev = "2f46ce4cc072107153da0cefe15bdc69aa5b84d0" }
\ No newline at end of file
+wasi-experimental-http-wasmtime = { git = "https://github.com/Mossaka/wasi-experimental-http", rev = "a4e1ee14a1f2d520289185537c64c315c87c1836" }
```
Comment (lann) [inline crates/outbound-http/Cargo.toml:22]: This is still pointing at your fork.

File: crates/outbound-http/src/lib.rs [hunk 4]
```diff
@@ -25,7 +27,8 @@ impl OutboundHttp {
     }
 
     /// Check if guest module is allowed to send request to URL, based on the list of
-    /// allowed hosts defined by the runtime.
+    /// allowed hosts defined by the runtime. If the list of allowed hosts contains
+    /// `*`, then all hosts are allowed.
```
Comment (lann) [inline crates/outbound-http/src/lib.rs:31]:
```suggestion
    /// the string `insecure:allow-all`, then all hosts are allowed.
```

**Review by lann — CHANGES_REQUESTED** [review]

File: crates/outbound-http/Cargo.toml (hunk 3 above)
Comment (Mossaka) [reply crates/outbound-http/Cargo.toml:22]: you're right!

**Review by lann — APPROVED** [review]

**Pull request merged**